mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/V1ToV2Migration.java

```diff
@@ -64,9 +64,15 @@ public V1ToV2Migration(CassandraMessageDAO messageDAOV1, CassandraMessageDAOV2 m
         this.migrationExecutor = Executors.newFixedThreadPool(cassandraConfiguration.getV1ToV2ThreadCount());
         boolean ensureFifoOrder = false;
         this.messagesToBeMigrated = new ArrayBlockingQueue<>(cassandraConfiguration.getV1ToV2QueueLength(), ensureFifoOrder);
-        IntStream.range(0, cassandraConfiguration.getV1ToV2ThreadCount())
-            .mapToObj(i -> new V1ToV2MigrationThread(messagesToBeMigrated, messageDAOV1, messageDAOV2, attachmentLoader))
-            .forEach(migrationExecutor::execute);
+        executeMigrationThread(messageDAOV1, messageDAOV2, cassandraConfiguration);
+    }
+
+    private void executeMigrationThread(CassandraMessageDAO messageDAOV1, CassandraMessageDAOV2 messageDAOV2, CassandraConfiguration cassandraConfiguration) {
+        if (cassandraConfiguration.isOnTheFlyV1ToV2Migration()) {
```

Comment:
No, if we don't want the _ on the fly migration_ in the configuration, then we don't have to create the associated threads.